admiral/pkg/controller/admiral/dependency_test.go

```diff
@@ -20,3 +23,76 @@ func TestNewDependencyController(t *testing.T) {
 		t.Errorf("Dependency controller should never be nil without an error thrown")
 	}
 }
+
+func TestGetDependencies(t *testing.T) {
+	stop := make(chan struct{})
+	handler := test.MockDependencyHandler{}
+
+	dependencyController, err := NewDependencyController(stop, &handler, "../../test/resources/admins@fake-cluster.k8s.local", "ns", time.Duration(1000))
+
+	if err != nil {
+		t.Errorf("Unexpected err %v", err)
+	}
+
+	if dependencyController == nil {
+		t.Errorf("Dependency controller should never be nil without an error thrown")
+	}
+
+	deps, _  := dependencyController.GetDependencies()
+
+	if deps != nil {
+		t.Errorf("deps is not nil")
+	}
+
+}
+
+func TestDependencyAddUpdateAndDelete(t *testing.T) {
+	stop := make(chan struct{})
+	handler := test.MockDependencyHandler{}
+
+	dependencyController, err := NewDependencyController(stop, &handler, "../../test/resources/admins@fake-cluster.k8s.local", "ns", time.Duration(1000))
+
+	if err != nil {
+		t.Errorf("Unexpected err %v", err)
+	}
+
+	if dependencyController == nil {
+		t.Errorf("Dependency controller should never be nil without an error thrown")
+	}
+
+	//test add
+	depName := "dep1"
+	dep := model.Dependency{IdentityLabel: "identity", Destinations:[]string{"greeting", "payments"}, Source: "webapp"}
+	depObj := makeK8sDependencyObj(depName, "namespace1", dep)
+	dependencyController.Added(depObj)
+
+	newDepObj := dependencyController.Cache.Get(depName)
+
+	if newDepObj == nil || newDepObj.Spec.Source != "webapp" {
+		t.Errorf("dep add failed")
+	}
+
+	//test update
+	updatedDep := model.Dependency{IdentityLabel: "identity", Destinations:[]string{"greeting", "payments", "newservice"}, Source: "webapp"}
+	dependencyController.Updated(makeK8sDependencyObj(depName, "namespace1", updatedDep), depObj)
+
+	updatedDepObj := dependencyController.Cache.Get(depName)
+
+	if updatedDepObj == nil || updatedDepObj.Spec.Source != "webapp" ||  len(updatedDepObj.Spec.Destinations) != 3 {
```

Comment:
You can use the `cmp` library for doing deep equals of objects - it could be useful here instead of comparing the different fields on their own

Reply:
Done.

Reply:
Done